compiler/rustc_resolve/src/late.rs

```diff
@@ -4304,6 +4304,13 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
             }
 
             ExprKind::Struct(ref se) => {
+                // FIXME(fmease): Add a comment maybe.
+                if se.path == kw::Empty
+                    && self.r.dcx().has_stashed_diagnostic(expr.span, StashKey::StructLitNoType)
+                {
+                    return;
```

Comment:
We're skipping name resolution entirely as mentioned in the PR description.
@petrochenkov, I'm sure this is super wrong. Please do enlighten me how I
could do better. No rush, though, this is a low priority experimental diagnostic PR.

Reply:
Do pings actually work inside draft PRs? :thinking: